kapitan/initialiser.py

```diff
@@ -20,27 +20,150 @@
+        self.targets = targets
+        self.compile_inputs = compile_inputs
+        current_pwd = os.path.dirname(__file__)
+        self.templates_directory = os.path.join(current_pwd, 'inputs/templates')
+
+        self.copy_path_list = []
+
+
+
+    # copy_tree(templates_directory, directory)
+    def get_target_template(self):
+        path = os.path.join(self.templates_directory, self.target_dir, self.target_template_name)
+        with open(path) as f:
+            obj = yaml.safe_load(f.read())
+        return obj
+
+    def get_component_template(self):
+        path = os.path.join(self.templates_directory, self.component_dir, self.component_template_name)
+        with open(path) as f:
+            obj = yaml.safe_load(f.read())
+        return obj
+
+    def generate_copy(self):
+        self.copy_inventory()
+        self.copy_target_file()
+        self.copy_component_file()
+        self.copy_compile_components()
+
+    def copy_inventory(self):
+        """
+            copy all files in inventory except my_target.yml
+        """
+        template_inventory = os.path.join(self.templates_directory,'inventory')
+        new_inventory = os.path.join(self.directory,'inventory')
+        shutil.copytree(template_inventory,new_inventory,
+                        ignore=shutil.ignore_patterns("my_target.yml"))
+
+    def copy_target_file(self):
+        """
+            copy target_file to new location with target-name provided by user
+            if is empty target-name then copy original file
+        """
+        target_file = self.get_target_template()
+        if len(self.targets):
+            for i in self.targets:
+                target_file['parameters']['target_name'] = i
+                path = self.get_new_target_path("%s.yml" % i)
+                self.dump_yaml(target_file,path)
+        else:
+            path = self.get_new_target_path(self.target_template_name)
+            self.dump_yaml(target_file, path)
+
+    def copy_component_file(self):
+        """
+            copy component file `my_component.yml` to new directory
+            only copy compile_obj's input_type is in compile-input
+            if compile-input is not provided my user copy all compile_obj
+            append input_paths of accepted compile_obj to copy_path_list
+        """
+
+        component_file = self.get_component_template()
+        compile_objs = []
+        for compile_obj in component_file['parameters']['kapitan']['compile']:
+            if compile_obj['input_type'] in self.compile_inputs or len(self.compile_inputs) == 0:
+                self.copy_path_list += compile_obj['input_paths']
+                compile_objs.append(compile_obj)
+        component_file['parameters']['kapitan']['compile'] = compile_objs
+
+        path = self.get_new_component_path()
+        self.dump_yaml(component_file, path)
+
+    def copy_compile_components(self):
+        """
+            copy for files present in copy_path_list to new location
```

Comment:
This comment needs rewording as well. Also the function is only used in one place. Does it need to be a function?